### PR TITLE
Rework saddle-points exercise

### PR DESCRIPTION
**Pull request by kytrinyx**
I really struggled with this, because I don't actually properly understand how saddle-points are used for optimization.

ChatGPT explained it, but it was wrong. I couldn't get it to actually explain how anything works, it just kept regurgitating the same (wrong) explanation.

Finally, we asked for suggestions on the forum and dreig came up with a brilliant suggestion.

I've rewritten this to reflect the basic idea in dreig's story.

<details>
<summary>ChatGPT notes</summary>

Saddle points in a matrix occur at the intersection of a row and a column with the minimum value in the row and the maximum value in the column. Here are some real-life situations where detecting saddle points in a matrix may be useful:

In economics, a saddle point may represent a situation where two companies have reached an equilibrium in terms of their production and pricing strategies.

In game theory, saddle points may indicate an optimal strategy for a player in a two-player zero-sum game.

In transportation planning, saddle points may represent optimal routes for vehicles that minimize both the distance traveled and the time taken.

In fluid dynamics, saddle points can help identify critical points in a fluid flow where turbulence or vortices may form.

In image processing, saddle points can be used to detect edges or corners in an image, which is useful for object recognition and tracking.

Overall, detecting saddle points in a matrix can provide insight into complex systems and help optimize strategies and solutions in various fields.

----

Prompt: Imagine you are working for a transportation company that needs to optimize its routes between various cities. Your company wants to minimize the time it takes for a vehicle to travel between cities while also minimizing the distance traveled. You have been tasked with developing a program that can help the company identify the optimal route between two cities.

Instructions: Your programming task is to develop a function that takes in a matrix representing the distances between cities and returns the indices of any saddle points in the matrix. A saddle point is an element in the matrix that is both a minimum in its row and a maximum in its column. Your function should be able to handle any sized matrix and return all saddle points in the matrix. You may use any programming language of your choice to solve this task. Remember to test your function thoroughly with various inputs to ensure it works correctly. Good luck!

----

Prompt: Imagine you are working for a video game company that is developing a two-player zero-sum game. Your company wants to create an AI player that can beat human players consistently. The AI player will need to use game theory to determine the optimal strategy for each move. You have been tasked with developing a program that can help the AI player analyze the game and determine the best strategy for each move.

Instructions: Your programming task is to develop a function that takes in the current state of the game and returns the optimal move for the AI player to make. The game is a two-player zero-sum game, meaning that the gain of one player is the loss of the other player. You can assume that the game state is represented as a matrix where each cell represents a possible move for the AI player. Your function should use game theory to analyze the matrix and determine the best move for the AI player to make. You may use any programming language of your choice to solve this task. Remember to test your function thoroughly with various inputs to ensure it works correctly. Good luck!

</details>

### CODE REVIEW

File: exercises/saddle-points/introduction.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Introduction
+
+You work for a transportation company that needs to improve the efficiency of their routes.
+
+You have been tasked with developing a program that can help the company identify routes that are problematic.
+These are locations where there is no optimal solution based purely on distance.
+The routes through these areas can then be analyzed by experts in the company, to find better solutions.
```
Comment (ErikSchierboom) [inline exercises/saddle-points/introduction.md:7]: I don't have any idea here to be honest. We could crowd/forum-source it maybe?

Comment (kytrinyx) [reply exercises/saddle-points/introduction.md:7]: That's a good idea. I'll start a conversation there.

**Review by ErikSchierboom — APPROVED** [review]
Lovely!

File: exercises/saddle-points/introduction.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+You are planning on building a tree house in the woods near your house so that you can watch the sun rise and set.
+
+You've obtained data from a local survey company that shows the heights of all the trees in each rectangular section of the map.
+
+You need to analyze each grid on the map to find the perfect tree for your tree house.
+
+The best tree will be the tallest tree compared to all the other trees to the east and west, so that you have the best possible view of the sunrises and sunsets.
+
+You don't like climbing too much, so the perfect tree will also be the shortest among all the trees to the north and to the south.
```
Comment (iHiD) [inline exercises/saddle-points/introduction.md:11]: I suggest not having each sentence as a paragraph here. Maybe L5+7 could be merged into one paragraph. And 9+11?

**Review by iHiD — APPROVED** [review]
This is great :)

**Pull request closed without merging**